tests/zeromq.terminal/Program.cs

```diff
@@ -14,67 +13,68 @@ namespace zeromq.terminal
     ///</summary>
     class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
             var version = args.FirstOrDefault();
             var configuration = SocketConfiguration.InprocConfig("this-inproc-sir-" + Guid.NewGuid());
             
             if (version != null)
-                RunPubSub(configuration);
+                await RunPubSub(configuration);
             else
-                RunReqRep(configuration);
+                await RunReqRep(configuration);
 
             NetMQConfig.Cleanup();
         }
 
-        private static void RunPubSub(SocketConfiguration configuration)
+        private static async Task RunPubSub(SocketConfiguration configuration)
         {
-            // only works with InProc
+            // it works only for InProc right now
             var pubSub = new PubSub(configuration);
-            Task.Run(() => 
+            await Task.Run(() => 
             {
                 System.Console.WriteLine("setting up the subscriber");
                 pubSub.SubscribeHandler<Message>(
                     (m) => System.Console.WriteLine("message came in: " + m.Text)
                 );
             });
             
-            Thread.Sleep(1000);
-            
             System.Console.WriteLine("now publishes");
             var input = "";
             while (input != "quit")
             {
                 System.Console.WriteLine("publish next one");
-                var xt = pubSub.PublishAsync(new Message() {Text = "publsihed!"}).Result;
+                var xt = await pubSub.PublishAsync(new Message() {Text = "publsihed!"});
                 System.Console.WriteLine(xt);
 
                 input = Console.ReadLine();
             }
         }
 
-        private static void RunReqRep(SocketConfiguration configuration)
+        private static async Task RunReqRep(SocketConfiguration configuration)
         {
             var socket = new Socket(configuration);
 
+            var cts = new CancellationTokenSource();
```

Comment:
`CancellationTokenSource` is `IDisposable` ... well, we don't leak any timer here, but IMO it's still good practice to have it disposed.